src/main/java/com/sysdig/jenkins/plugins/sysdig/SysdigIaCScanBuilder.java

```diff
@@ -0,0 +1,350 @@
+            sysdigEnv = "https://secure-staging.sysdig.com";
+        }
+        cmd.add(sysdigEnv);
+
+        if (isRecursive) {
+            cmd.add("-r");
+        }
+
+        if (listUnsupported) {
+            cmd.add("--list-unsupported-resources");
+        }
+
+        severity(cmd);
+
+        cmd.add(path);
+
+        return cmd;
+    }
+
+    private void severity(Vector<String> cmd) {
+        cmd.add("-f");
+        cmd.add(severityThreshold);
+    }
+
+    @Override
+    public void perform(Run<?, ?> run, FilePath workspace, Launcher launcher, TaskListener listener)
+            throws InterruptedException, IOException {
+        
+    }
+
+    @Override
+    public boolean perform(Build<?, ?> build, Launcher launcher, BuildListener listener)
+            throws InterruptedException, IOException {
+        // TODO Auto-generated method stub
+        super.perform(build, launcher, listener);
+
+        CLIDownloadAction act = null;
+        try {
+            listener.getLogger().println("trying to download cli");
+            String cwd = System.getProperty("user.dir");
+            act = new CLIDownloadAction("IaC scanner", cwd,version);
+        } catch (Exception e) {
+            // TODO Auto-generated catch block
+            listener.getLogger().println("failed to download cli");
+
+            e.printStackTrace();
+
+            listener.error("failed:%s", e.getMessage());
+            return false;
+        }
+        build.addAction(act);
+        listener.getLogger().println(act.chwd());
+
+        listener.getLogger().println("starting to scan");
+        try {
+            if (act.chwd() == "") {
+                listener.error("failed empty path");
+
+                throw new Exception("empty path");
+            }
+            String exec = act.chwd();
+            ProcessBuilder pb = new ProcessBuilder(buildCommand(exec));
+            Map<String, String> envv = pb.environment();
+            envv.put("SECURE_API_TOKEN", secureAPIToken);
+
+            listener.getLogger().println(pb.command());
+
+            Process p = pb.start();
+            listener.getLogger().println("started...");
+
+            String output = getProcessOutput(p);
+            int exitCode = p.exitValue();
+            listener.getLogger().printf("finished status %d\n", exitCode);
+
+            listener.getLogger().printf("%s", output);
+
+            switch (exitCode) {
+                case 1:
+                    throw new FailedCLIScan(String.format("scan failed \n\n%s", output));
+
+                case 2:
+                    throw new FailedCLIScan(String.format("scan failed \n\n%s", output));
+                default:
+
+                    break;
+            }
+
+        } catch (FailedCLIScan e) {
+            listener.error("iac scan %s", e.getMessage());
+            return false;
+        } catch (BadParamCLIScan e) {
+            listener.error("iac scan %s", e.getMessage());
+            return false;
```

Comment:
Since we have different classes for the different exceptions, can we give the user more information about what failed? Both exceptions seem to be the same error from the customer's POV.